### PR TITLE
fix: don't depend on the GitHub API to check release

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/buildx/pull/1563

We are currently using the GitHub API to check for latest and tagged releases to make sure they exist before download. But this requires using a token to avoid rate-limit. It's fine for public runners but GHES runners [don't have the `github.token` populated](https://github.com/goreleaser/goreleaser-action/blob/9754a253a8673b0ea869c2e863b4e975497efd0c/action.yml#L29-L37) automatically (see https://github.com/docker/setup-buildx-action/issues/194). They need to create a PAT.

We actually don't need to use the GitHub API at all as we have the list of releases available:
* https://goreleaser.com/static/releases.json
* https://goreleaser.com/static/releases-pro.json
* https://goreleaser.com/static/latest
* https://goreleaser.com/static/latest-pro

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,10 +23,10 @@
   "dependencies": {
     "@actions/core": "^1.10.0",
     "@actions/exec": "^1.1.1",
-    "@actions/github": "^5.1.1",
     "@actions/http-client": "^2.0.1",
     "@actions/tool-cache": "^2.0.1",
     "js-yaml": "^4.1.0",
+    "semver": "^7.3.8",
```
Comment (crazy-max) [inline package.json:29]: Also found that `semver` module was not properly registered.

File: package.json [hunk 2]
```diff
@@ -23,10 +23,10 @@
   "dependencies": {
     "@actions/core": "^1.10.0",
     "@actions/exec": "^1.1.1",
-    "@actions/github": "^5.1.1",
```
Comment (crazy-max) [inline package.json:26]: This module is not necessary

**Review by caarlos0 — APPROVED** [review]
do you think this should be a v4 release?

**Comment by crazy-max** [comment]
@caarlos0 I don't think it's necessary. Introducing `github_token` should have been a `v4` but we are going back to what we had before so I think it's fine to create a minor release.

**Comment by caarlos0** [comment]
> @caarlos0 I don't think it's necessary. Introducing `github_token` should have been a `v4` but we are going back to what we had before so I think it's fine to create a minor release.

ah, right

lgtm then :D

**Comment by crazy-max** [comment]
Oh sorry I misunderstood, I thought you were saying a new major (v5) so yeah we shoud do a release

**Comment by crazy-max** [comment]
`v4.2.0` and `v4`

**Pull request merged**